src/in-memory-backend.service.ts

```diff
@@ -533,9 +534,9 @@ export class InMemoryBackendService {
   }
 
   protected getLocation(href: string) {
-    const l = document.createElement('a');
-    l.href = href;
-    return l;
+    // default the base url when not running in browser
+    let base = document ? document.location.href : 'http://node';
+    return new URL(href, base);
```

Comment:
Works in Chrome, Firefox, and Edge, but fails in IE 11.  I'll work on an alternative.